### PR TITLE
Adding latest functional protocol changes and python decoder to dev

### PR DESCRIPTION
**Pull request by zaitera**

### CODE REVIEW

File: .flutter-plugins-dependencies [hunk 1]
```diff
@@ -1 +1 @@
...
...vider","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-0.4.1/","dependencies":[]},{"name":"permission_handler","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/","dependencies":[]}],"android":[{"name":"e2e","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/e2e-0.2.4+4/","dependencies":[]},{"name":"flutter_blue","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.7.2/","dependencies":[]},{"name":"path_provider","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-0.4.1/","dependencies":[]},{"name":"permission_handler","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/","dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"e2e","dependencies":[]},{"name":"flutter_blue","dependencies":[]},{"name":"path_provider","dependencies":[]},{"name":"permission_handler","dependencies":[]}],"date_created":"2020-11-02 20:04:43.355536","version":"1.20.2"}
```
Comment (lschiavini) [inline .flutter-plugins-dependencies]: This file shouldn't be updated in the dev rep.

File: .flutter-plugins [hunk 2]
```diff
@@ -1,3 +1,5 @@
 # This is a generated file; do not edit or check into version control.
 e2e=/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/e2e-0.2.4+4/
 flutter_blue=/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.7.2/
+path_provider=/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-0.4.1/
+permission_handler=/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/
```
Comment (lschiavini) [inline .flutter-plugins]: This file also shouldn't be updated in dev.

File: .flutter-plugins-dependencies [hunk 3]
```diff
@@ -1 +1 @@
...
...vider","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-0.4.1/","dependencies":[]},{"name":"permission_handler","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/","dependencies":[]}],"android":[{"name":"e2e","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/e2e-0.2.4+4/","dependencies":[]},{"name":"flutter_blue","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/flutter_blue-0.7.2/","dependencies":[]},{"name":"path_provider","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/path_provider-0.4.1/","dependencies":[]},{"name":"permission_handler","path":"/home/abdullah/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/permission_handler-5.0.1+1/","dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"e2e","dependencies":[]},{"name":"flutter_blue","dependencies":[]},{"name":"path_provider","dependencies":[]},{"name":"permission_handler","dependencies":[]}],"date_created":"2020-10-30 17:02:13.051116","version":"1.20.2"}
```
Comment (lschiavini) [inline .flutter-plugins-dependencies]: Another auto generated file that shouldn't be in source control

**Review by lschiavini — COMMENTED** [review]
Needs checking some files on .gitignore

File: .flutter-plugins-dependencies (hunk 1 above)
Comment (zaitera) [reply .flutter-plugins-dependencies]: apparently it was being tracked since before on dev

File: .flutter-plugins (hunk 2 above)
Comment (zaitera) [reply .flutter-plugins]: apparently it was being tracked since before on dev

File: .flutter-plugins-dependencies (hunk 3 above)
Comment (zaitera) [reply .flutter-plugins-dependencies]: apparently it was being tracked since before on dev

**Comment by lschiavini** [comment]
Good to go

**Review by lschiavini — APPROVED** [review]
Good to go

**Pull request merged**